Makefile

```diff
@@ -1,565 +1,12 @@
-COREWEBAPPS_DIR?=$(PROFILE_DIR)
-
-define BUILD_CONFIG
-{ \
-  "ADB" : "$(patsubst "%",%,$(ADB))", \
-  "GAIA_DIR" : "$(GAIA_DIR)", \
-  "PROFILE_DIR" : "$(PROFILE_DIR)", \
-  "PROFILE_FOLDER" : "$(PROFILE_FOLDER)", \
-  "COREWEBAPPS_DIR" : "$(COREWEBAPPS_DIR)", \
-  "GAIA_SCHEME" : "$(SCHEME)", \
-  "GAIA_DOMAIN" : "$(GAIA_DOMAIN)", \
-  "DEBUG" : "$(DEBUG)", \
-  "LOCAL_DOMAINS" : "$(LOCAL_DOMAINS)", \
-  "DESKTOP" : "$(DESKTOP)", \
-  "DEVICE_DEBUG" : "$(DEVICE_DEBUG)", \
-  "NO_LOCK_SCREEN" : "$(NO_LOCK_SCREEN)", \
-  "SCREEN_TIMEOUT" : "$(SCREEN_TIMEOUT)", \
-  "SYSTEM" : "$(SYSTEM)", \
-  "GAIA_PORT" : "$(GAIA_PORT)", \
-  "GAIA_LOCALES_PATH" : "$(GAIA_LOCALES_PATH)", \
-  "GAIA_INSTALL_PARENT" : "$(GAIA_INSTALL_PARENT)", \
-  "LOCALES_FILE" : "$(subst \,\\,$(LOCALES_FILE))", \
-  "GAIA_KEYBOARD_LAYOUTS" : "$(GAIA_KEYBOARD_LAYOUTS)", \
-  "GAIA_KEYBOARD_PRELOAD_DICT_LAYOUTS" : "$(GAIA_KEYBOARD_PRELOAD_DICT_LAYOUTS)", \
-  "GAIA_KEYBOARD_ENABLE_USER_DICT" : "$(GAIA_KEYBOARD_ENABLE_USER_DICT)", \
-  "LOCALE_BASEDIR" : "$(subst \,\\,$(LOCALE_BASEDIR))", \
-  "BUILD_APP_NAME" : "$(BUILD_APP_NAME)", \
-  "PRODUCTION" : "$(PRODUCTION)", \
-  "GAIA_OPTIMIZE" : "$(GAIA_OPTIMIZE)", \
-  "GAIA_DEVICE_TYPE" : "$(GAIA_DEVICE_TYPE)", \
-  "GAIA_DEV_PIXELS_PER_PX" : "$(GAIA_DEV_PIXELS_PER_PX)", \
-  "DOGFOOD" : "$(DOGFOOD)", \
-  "OFFICIAL" : "$(MOZILLA_OFFICIAL)", \
-  "GAIA_DEFAULT_LOCALE" : "$(GAIA_DEFAULT_LOCALE)", \
-  "GAIA_PRETRANSLATE" : "$(GAIA_PRETRANSLATE)", \
-  "GAIA_CONCAT_LOCALES" : "$(GAIA_CONCAT_LOCALES)", \
-  "GAIA_ENGINE" : "xpcshell", \
-  "GAIA_DISTRIBUTION_DIR" : "$(GAIA_DISTRIBUTION_DIR)", \
-  "GAIA_APPDIRS" : "$(GAIA_APPDIRS)", \
-  "GAIA_ALLAPPDIRS" : "$(GAIA_ALLAPPDIRS)", \
-  "GAIA_MEMORY_PROFILE" : "$(GAIA_MEMORY_PROFILE)", \
-  "NOFTU" : "$(NOFTU)", \
-  "REMOTE_DEBUGGER" : "$(REMOTE_DEBUGGER)", \
-  "TARGET_BUILD_VARIANT" : "$(TARGET_BUILD_VARIANT)", \
-  "SETTINGS_PATH" : "$(subst \,\\,$(SETTINGS_PATH))", \
-  "FTU_PING_URL": "$(FTU_PING_URL)", \
-  "KEYBOARD_LAYOUTS_PATH" : "$(KEYBOARD_LAYOUTS_PATH)", \
-  "CONTACTS_IMPORT_SERVICES_PATH" : "$(CONTACTS_IMPORT_SERVICES_PATH)", \
-  "EMAIL_SERVICES_PATH" : "$(EMAIL_SERVICES_PATH)", \
-  "STAGE_DIR" : "$(STAGE_DIR)", \
-  "GAIA_APP_TARGET" : "$(GAIA_APP_TARGET)", \
-  "BUILD_DEBUG" : "$(BUILD_DEBUG)", \
-  "VARIANT_PATH" : "$(VARIANT_PATH)", \
-  "REBUILD": "$(REBUILD)", \
-  "P" : "$(P)", \
-  "VERBOSE" : "$(VERBOSE)", \
-  "RAPTOR" : "$(RAPTOR)", \
-  "SHARE_PERF_USAGE": "$(SHARE_PERF_USAGE)", \
-  "DEFAULT_KEYBOAD_SYMBOLS_FONT": "$(DEFAULT_KEYBOAD_SYMBOLS_FONT)", \
-  "DEFAULT_GAIA_ICONS_FONT": "$(DEFAULT_GAIA_ICONS_FONT)" \
-}
-endef
-
-export BUILD_CONFIG
-
-# Generate profile/
-$(PROFILE_FOLDER): profile-dir build-app test-agent-config contacts extensions b2g_sdk .git/hooks/pre-commit
-ifeq ($(BUILD_APP_NAME),*)
-	@echo "Profile Ready: please run [b2g|firefox] -profile $(CURDIR)$(SEP)$(PROFILE_FOLDER)"
-endif
+-include all.mk
```

Comment:
Could we just trigger all.mk and pass MAKEFLAGS into it rather than include it here?